code/game/machinery/shuttle/shuttle_heater.dm

```diff
@@ -170,3 +190,261 @@
+		. += "The engine heater's screws seem <b>loose</b>."
+	if(damage_state == DAMAGE_NONE && pressure_damage < PRESSURE_DAMAGE_MAX / 2)
+		. += "The engine heater is in good condition."
+
+/obj/machinery/atmospherics/components/unary/shuttle/fire_heater/proc/return_gas()
+	var/datum/gas_mixture/air_contents = airs[1]
+	if(!air_contents)
+		return
+	else
+		return air_contents.return_pressure()
+
+/obj/machinery/atmospherics/components/unary/shuttle/fire_heater/proc/return_gas_capacity()
+	var/datum/gas_mixture/air_contents = airs[1]
+	if(!air_contents)
+		return
+	return air_contents.return_volume()
+
+/obj/machinery/atmospherics/components/unary/shuttle/fire_heater/proc/update_gas_stats()
+	var/datum/gas_mixture/air_contents = airs[1]
+	if(!air_contents)
+		return
+	air_contents.set_volume(gas_capacity)
+	air_contents.set_temperature(T20C)
+
+/obj/machinery/atmospherics/components/unary/shuttle/fire_heater/proc/has_fuel(required, datum/gas/gas_type)
+	var/datum/gas_mixture/air_contents = airs[1]
+	if(!air_contents)
+		return
+	return air_contents.get_moles(gas_type) >= required
+
+/**
+ * consumes a portion of the mols and checks how much could combust to make thrust.
+ * oxidation_power is the total value of all the oxidizers
+ * fuel_power is ^ but for fuel
+ */
+/obj/machinery/atmospherics/components/unary/shuttle/fire_heater/proc/consume_fuel(gas_consumed)
+	var/datum/gas_mixture/air_contents = airs[1]
+	if(!air_contents)
+		return
+
+	else
+		var/oxidation_power = 0
+		var/fuel_power = 0
+		var/thrust_power = 0
+		var/gas_amount = 0
+
+		for(var/id in air_contents.get_gases())
+			gas_amount = air_contents.get_moles(id) * gas_consumed //this takes a percent (set by gas_consumed) and multiplies it by the total gas to get the amount of gas used by the calculation.
+
+			// adds each oxidizer's power to the total oxidation max
+			if(id == GAS_O2)
+				oxidation_power += O2_OXIDATION_VALUE * gas_amount
+			if(id == GAS_NITRYL)
+				oxidation_power += NITRYL_OXIDATION_VALUE * gas_amount
+			if(id == GAS_NITROUS) //burning nitrous cools down the heater's main tank, just like it cools the intake on real cars.
+				oxidation_power += NITROUS_OXIDATION_VALUE * gas_amount
+				var/heat_capacity = gas_amount * NITROUS_COOLING_MULTIPIER
+				var/air_heat_capacity = air_contents.heat_capacity()
+				var/combined_heat_capacity = heat_capacity + air_heat_capacity
+				if(combined_heat_capacity > 0)
+					var/combined_energy = heat_capacity * NITROUS_COOLING_MIN + air_heat_capacity * air_contents.return_temperature()
+					air_contents.set_temperature(combined_energy/combined_heat_capacity)
+
+			// adds each fuel gas's power to the fuel max (air.get_fuel_amount is busted, and trit should be Better anyways.)
+			if(id == GAS_PLASMA)
+				fuel_power += PLASMA_THRUSTER_VALUE * gas_amount
+			if(id == GAS_TRITIUM)
+				fuel_power += TRITRIUM_THRUSTER_VALUE * gas_amount
+			if(id == GAS_HYDROGEN)
+				fuel_power += HYDROGEN_THRUSTER_VALUE * gas_amount
```

Comment:
switch statement would work well here

Reply:
ok! Ill figure that out tomorrow I think